src/settings/BatchGroupConfigurationSettings/BatchGroupConfigurationSettings.js

```diff
@@ -31,14 +44,94 @@ const BatchGroupConfigurationSettings = ({ mutator }) => {
   // eslint-disable-next-line react-hooks/exhaustive-deps
   []);
 
-  const onSaveExportConfig = useCallback(
-    () => {
+  const fetchExportConfig = useCallback(
+    (id) => {
+      setIsLoading(true);
+      setExportConfig();
+      setCredentials();
 
+      const setConfigsAndFetchCreds = (configs) => {
+        if (configs.length) {
+          const exportConfigId = configs[0].id;
+
+          setExportConfig(configs[0]);
+          mutator.exportConfigId.update({ id: exportConfigId });
+
+          return mutator.credentials.GET();
+        } else {
+          setExportConfig({});
+
+          return Promise.resolve();
+        }
+      };
+
+      const setCreds = (creds) => {
```

Comment:
could you please describe the reason of extracting callbacks to the separate vars?

Reply:
I thought it was your suggestion to make it simple